Fixed bug related to changing currencies

papasnowee opened the pull request:
## Links
[TASK](https://www.notion.so/DASH-Fixed-bug-related-to-changing-currencies-0fabe6837235414f9c0fd5c36a4bfc54)

## Description
if you change the value of currencies many times, the application will crash until the localstorage is cleared.

### Resolve
 The default value of currencies '1' --> 'USD'. Added memoize function for creating options.
![bug currencie-change](https://user-images.githubusercontent.com/34766620/113474147-c6f61580-9476-11eb-932c-6bc23c6285e7.jpg)
![bug](https://user-images.githubusercontent.com/34766620/113474153-cfe6e700-9476-11eb-8f5a-63f83d89b742.jpeg)

cryptopower80 commented:
I'm not able to reproduce the issue. Could you please provide the correct steps?

cryptopower80 commented on src/components/userSettings/currency/Currency.jsx:
```diff
@@ -1,40 +1,38 @@
+    window.localStorage.setItem('HarvestFinance:currency', e.target.value);
+  };
+
+  const options = useMemo(() => {
```
I think we should try to avoid using `useMemo` and `memo` wherever possible and find alternative optimizations. `exchangeRates` should be re-generated just in case the value is different. I think we could optimize that part instead of this component. However, we should use `memo()` since it's a component and not just a function

papasnowee replied:
This optimization avoids re-rendering of the options array. The Render operation is more complex than creating a useMemo. Maybe I didn't understand you.

cryptopower80 replied:
Yes, but options are html elements and in this case it's necessary to use `memo()` since it is for components.

papasnowee replied:
Ok, you are right. Corrected It.

papasnowee commented:
> I'm not able to reproduce the issue. Could you please provide the correct steps?

Change currency value 20-30 or more times please and you will crash the app. I changed the currency 

> I'm not able to reproduce the issue. Could you please provide the correct steps?

Please change the value of the currency 20-30 times. I changed the value of the currencies in order.
